.github/workflows/build_pyspinw.yml

```diff
@@ -103,6 +103,14 @@ jobs:
         run: |
           cd ${{ github.workspace }}/python
           python -m pip wheel --no-deps --wheel-dir build .
+      - name: Run python test
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          pip install scipy
+          cd ${{ github.workspace }}/python
+          pip install build/*whl
+          cd tests
+          python -m unittest
```

Comment:
Ah yes, I think that if the tests fail the wheels will not be built (we want this, right?). 

Actually there's a mistake in the yaml... line 107 `if: matrix.os == 'ubuntu-latest'` should be removed... this stage only runs on `ubuntu-latest` and doesn't have a `matrix` so the `if` always evaluates to false and this step never runs...

I was also going to port the `systemtest_spinwave` class to Python to allow the rest of the system tests to be ported easily but haven't had time... that might go into a different PR...

Reply:
> Ah yes, I think that if the tests fail the wheels will not be built (we want this, right?).

I agree!

> Actually there's a mistake in the yaml... line 107 `if: matrix.os == 'ubuntu-latest'` should be removed... this stage only runs on `ubuntu-latest` and doesn't have a `matrix` so the `if` always evaluates to false and this step never runs...

Good spot

> I was also going to port the `systemtest_spinwave` class to Python to allow the rest of the system tests to be ported easily but haven't had time... that might go into a different PR...

Yes I thought about this, but it might be overkill if we just want to check the python interface works and returns the correct object (and leave the detailed validation in matlab). I thought if we can use the same `.mat` files then we can get away with not needing to generate reference data if it fails etc. - however, atm the `.mat` file doesn't seem to agree with the output object! I'll try and look into this today